src/main/java/io/knotx/junit5/KnotxExtension.java

```diff
@@ -250,33 +241,32 @@ private void loadKnotxConfig(Vertx vertx, KnotxApplyConfiguration knotxConfig) {
   }
 
   private DeploymentOptions createConfig(String[] paths) {
-    JsonObject storesConfig = new ConfigRetrieverOptions()
-        .setStores(
-            Stream.of(paths)
-                .map(path ->
-                    new ConfigStoreOptions()
-                        .setType("file")
-                        .setFormat(getConfigFormat(path))
-                        .setConfig(new JsonObject().put("path", path)))
-                .collect(Collectors.toList())
-        )
-        .toJson();
+    ConfigRetrieverOptions retrieverOptions = new ConfigRetrieverOptions();
+
+    Stream.of(paths).forEach(this::guardConfigFormat);
+
+    JsonObject config = new JsonObject();
+
+    config.put("paths", Arrays.asList(paths));
+    config.put("overrides", wiremockExtension.getConfigOverrides());
+
+    retrieverOptions.addStore(
+        new ConfigStoreOptions()
+            .setType("json")
+            .setFormat("knotx")
+            .setOptional(false)
+            .setConfig(config));
+
+    JsonObject storesConfig = retrieverOptions.toJson();
     return new DeploymentOptions()
-        .setConfig(
-            new JsonObject()
-                .put(
-                    "configRetrieverOptions",
-                    storesConfig));
+        .setConfig(new JsonObject().put("configRetrieverOptions", storesConfig));
   }
 
-  private String getConfigFormat(String path) {
+  private void guardConfigFormat(String path) {
     String extension = path.substring(path.lastIndexOf('.') + 1);
-    if ("conf".equals(extension)) {
```

Comment:
Why not hocon?

Reply:
Moved string literals to more explaining constants.